accessibility/common-remediations.md

```diff
@@ -0,0 +1,213 @@
+
+All elements that receive focus MUST have visible focus styles. 
+
+Browser default focus styles are likely insufficient as they can be difficult to see in some designs. We prefer focus styles to be highly visible. 
+
+
+We do this:
+```css
+a:focus {
+   outline: 3px solid #fece3e;
+}
+
+a:focus, button:focus {
+   background-color: #fc0;
+   outline: 2px solid #fc0;
+}
+```
+
+We _don't_ do this:
+```css
+a:focus {
+   outline: 0;
+}
+```
+
+
+### Unsemantic elements used for interactive components
+
+Built-in interactive HTML elements like anchors, text fields, buttons, and select lists (among others) are handled automatically by the browser. Users can interact with them with the mouse, the keyboard, by voice, or any other kind of technology that they like. 
+
+Elements with no semantics (divs and spans) don’t have this automatic handling, and a simple `onclick` handler doesn’t provide it. These elements generally can’t be interacted with by users of Assistive Technology unless you write large amounts of JavaScript to make it possible. 
+
+Where a built-in interactive element is available, use it. 
+
+We do this:
+```html
+<button>Toggle UI</button>
+<a href=”/”>Visit this location</a>
+```
+We _don’t_ do this:
+```html
+<div onclick=”doFunc();”>Toggle UI</div>
+<span>Visit this location</a>
+```
+
+
+### Links with no href
+
+All anchor links (the `a` element) require an `href` attribute in order to be functional as links. Omitting the `href` attribute prevents keyboard users and voice input users from being able to use the link at all. 
+
+We do this:
+```html
+<a href=”#location”>Visit this location</a>
+```
+We _don’t_ do this:
+```html
+<a onclick=”myFunc();”>Visit this location</a>
+```
+
+
+
+###  Inappropriate negative tabindex
+
+Interactive elements with a negative tabindex attribute (`tabindex=-1`) are not usable by keyboard users. 
+
+
+
+### Visual order is not followed with source code
+Just try to follow the visual order of elements while coding in HTML, don’t adjust elements to take positions with CSS, then this is confusing to keyboard users who expect to tab forwards and backwards in a linear sequence.
+
+As a rule, try tabbing through your pages every so often just to make sure you haven't accidentally messed up the tab order. 
+
+We do this:
+```html
+<button> I Should </button>
+<button> Be focused</button>
+<button> Last </button>
+```
+
+We don't do this:
+```html
+<button style=”float: right”> I Should </button>
+<button> Be focused</button>
+<button> Last </button>
+```
+
+
+
+## No advance warning when opening new window 
+
+Opening new windows automatically when a link is activated can be disorienting for people who have difficulty perceiving visual content, and for some people with cognitive disabilities, if they are not warned in advance.
+
+We do this:
+```html
+<a href="https://mysite.com" target="_blank" rel="noopener">
+ My site
+ <span class="visuallyhidden">, opens in a new window</span>
```

Comment:
Let's change this suggestion. A visually-hidden label doesn't do anything for sighted users, and sighted users can be confused by unexpected context changes too. 

The [a11y checklist recommends](https://github.com/springernature/frontend-playbook/blob/master/accessibility/accessibility-checklist.md#avoid-opening-links-in-new-windows) either a text label or an icon, so let's bring this remediation into line with that. 

I'm also going to open a PR on the checklist to clarify that the text label should be visible, as it's not clear enough there!